### ISSUE TITLE
Warning "wrong formatting options for api/mqtt enum format"

### ISSUE THREAD

**Issue by harvey637** [tp1de/ioBroker.ems-esp#58]
Hi,
running ems-esp-3.7.0-dev4 (and before) with enum settings as "Index".
Now with iobroker.emsesp v3.4.0 this warnung about wrong use of this setting, so I changed into "Value" to get a green adapter status.

I am not using km200.

In my opinion the setting "use enums as INDEX" make perfect sense for an API, as "Value" is language dependent.

This is the output with "Value":
```
  "hpcompon": 0,
  "hpactivity": "Abtauen",
  "hpswitchvalve": 1,
...  
  "hp4way": "KÃ¼hlen & Abtauen",
  "auxheaterstatus": 0,
  "auxheatrmode": "Eco",
  "auxheatmix": 0,
...
  "hppumpmode": "auto",

```
this is the fetch with enums as "Index"
```
  "hpcompon": 0,
  "hpactivity": 1,
  "hpswitchvalve": 1,
...
  "hp4way": 0,
  "auxheaterstatus": 0,
  "auxheatrmode": 0,
  "auxheatmix": 0,
...
  "hppumpmode": 0,

```

![grafik](https://github.com/tp1de/ioBroker.ems-esp/assets/10156426/26f22bb6-7894-4dbb-b740-17552b626dd3)

My expected behaviour is to transport easy formats in API/MQTT ....
In the object states ther is the translation of index into value:
```
{
  "_id": "ems-esp.0.boiler.hpactivity",
  "type": "state",
  "common": {
    "id": "boiler.hpactivity",
    "name": "ems: Kompressor-Betriebsmodus",
    "type": "mixed",
    "read": true,
    "write": false,
    "role": "value",
    "states": {
      "0": "keine",
      "1": "Heizen",
      "2": "Kühlen",
      "3": "Warmwasser",
      "4": "Pool",
      "5": "Unbekannt",
      "6": "Abtauen"
    },
```

So my whish is to keep the "correct" setting of api/mqtt settings for enums as index. And dont throw this as warning, the status color of the iobroker-emsesp should be GREEN (not yellow).

thank you for looking into this!
cu
Harvey
PS: until updated I need to stay at "index" with yellow adapter flag .... as some scripts also parse the api-call and fixing "hpactivity": "Abtauen" into a graph seems not beeing helpfull:
![grafik](https://github.com/tp1de/ioBroker.ems-esp/assets/10156426/d2cd2c7e-a7bd-4195-bf65-02ea05b07ef3)

(see te orange background for heating and WP Aktivität for none/heating/defrost .... textual representation seems useless)

**Comment by tp1de** [comment]
The checks for boolean and enum are according to the existing documentation. (since > 2 years)
When not passed polling is stopped in adapter version 3.4.0 because I had too many people not reading docu. 
But I understand your argument.

Please reinstall version 3.3.0. 
I need to check what to do.

**Comment by harvey637** [comment]
For me it looks, that the GUI always uses the VALUE for enums.
As seen in the iobroker objects I see (and saw since long time) that the enums have the textual representation in the iobroker-object, so they are fetched from the api-call when a object is created.

What is the use case for reading the value for enums and not the index?
Just asking to understand ....

(as you already know them without setting enums=VALUE)

**Comment by tp1de** [comment]
3.4.1 is out without the check for enum settings. 

I recognized that I was already accepting both ways within the code since a long time to avoid errors.
Just bool 0/1 is mandatory.

The ems-esp gateway way of creating objects was made similar to the output of km200 api and objects created by the old km200 adapter.

While testing again, I recognized that the last admin adapter version (6.17.6)  does not show the enum values anymore within the  object browser. Other users already opened GH issues.

Version 3.7 firmware is changing names and structure of warm water objects unless you use km200 structure. Be careful.
